### PR TITLE
fix build with musl libc

### PR DESCRIPTION
**Pull request by aidgamesa**

### CODE REVIEW

File: Userland/Libraries/LibCore/Group.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ ErrorOr<String> Group::generate_group_file() const
     ScopeGuard grent_guard([] { endgrent(); });
     setgrent();
     errno = 0;
-#ifndef AK_OS_MACOS
+#if !defined(AK_OS_MACOS) && defined(__GLIBC__)
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/Group.cpp:26]: This isn't quite sufficient. The check for !AK_OS_MACOS implicitly means (AK_OS_LINUX || AK_OS_SERENITY || AK_OS_OPENBSD || AK_OS_FREEBSD || .... etc ). Any checks for glibc should be asking AK_OS_LINUX && glibc.

**Comment by aidgamesa** [comment]
superbuild passed...
now i build serenityos image...

**Comment by aidgamesa** [comment]
superbuild passed successful
verdict: "sys/cdefs.h is deprecated" warning crash build if I comment its build!
now i'm build serenityos image...

**Comment by aidgamesa** [comment]
screenshot
its booted successful!

**Comment by aidgamesa** [comment]
fix sys/cdefs.h build error

**Review by linusg — COMMENTED** [review]
Listing that many categories is a bit silly, just use `Everywhere:` please :)

**Review by ADKaster — CHANGES_REQUESTED** [review]
Some other PRs have come along to try to address the same problem as part of https://github.com/SerenityOS/serenity/issues/16463

The only 3 left are 
- remove sys/cdefs.h entirely
- zero init msghdr and then set the fields we want in ntpquery.cpp
- unfortunate overriding of `__assert_fail` in test262 runner.

The last one is the only reason we would want AK_MUSL or a similar define. It might be that we want to use a SIGABRT handler instead to do this error reporting. that would likely be less undefined-behavior/less annoying for C library authors if we did that.

**Comment by ADKaster** [comment]
All of the mentioned changes have made it into master, feel free to open another PR if we break musl builds again!

**Pull request closed without merging**